app/models/course/assessment/answer/forum_post_response.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+class Course::Assessment::Answer::ForumPostResponse < ApplicationRecord
+  acts_as :answer, class_name: Course::Assessment::Answer.name
+
+  has_many :postpacks, class_name: Course::Assessment::Answer::ForumPost.name,
+                       dependent: :destroy, foreign_key: :answer_id, inverse_of: :answer
+
+  def assign_params(params)
+    acting_as.assign_params(params)
+    self.answer_text = params[:answer_text] if params[:answer_text]
+
+    return unless params[:selected_postpacks]
+
+    destroy_previous_selection
+
+    params[:selected_postpacks].each do |selected_postpack|
+      postpack = postpacks.new
+
+      postpack.forum_topic_id = selected_postpack[:topic][:id]
+
+      postpack.post_id = selected_postpack[:corePost][:id]
+      postpack.post_text = selected_postpack[:corePost][:text]
+      postpack.post_creator_id = selected_postpack[:corePost][:creatorId]
+      postpack.post_updated_at = selected_postpack[:corePost][:updatedAt]
+
+      if selected_postpack[:parentPost]
+        postpack.parent_id = selected_postpack[:parentPost][:id]
+        postpack.parent_text = selected_postpack[:parentPost][:text]
+        postpack.parent_creator_id = selected_postpack[:parentPost][:creatorId]
+        postpack.parent_updated_at = selected_postpack[:parentPost][:updatedAt]
+      end
+
+      postpack.save!
+    end
+  end
+
+  def get_postpacks
+    postpacks.each do |selected_post|
```

Comment:
> Hi @aylz, great job with this PR! Generally well done.
> 
> I've left comments where relevant, do look through and make the necessary fixes.
> 
> Here are some general comments as well:
> 
> * I strongly recommend going with post_packs / postPacks instead of having it as a single word postpacks. More semantic.
> * The React component names are too similar. Can we make it more explicit - `Forum` - forum what? What exactly is this component? Is it a drawer, is it a tab, is it a container, etc.
> * Duplication - can I check if duplication of forum post response questions has been handled?
> * JS files - need to reformat all files. Also, you might want to fix up your IDE settings.

Thanks @zhuhanming. Roger on your comments, and also noted on the naming (suffixing with `_post`) of the data fields. Will be using prettier for JS formatting instead of eslint which I'm currently using.